js/main.js

```diff
@@ -1,2 +1,32 @@
 /* global data */
 /* exported data */
+
+function handleInput(event) {
+  $image.src = event.target.value;
+}
+
+function handleSubmit(event) {
+  event.preventDefault();
+  var newEntry = {};
+  var newObj = {
+    title: $entryForm.elements.title.value,
+    imageURL: $entryForm.elements.photourl.value,
+    notes: $entryForm.elements.notes.value
+  };
+  for (var i in newObj) {
+    newEntry[i] = newObj[i];
+  }
+  newEntry.entryId = data.nextEntryId;
+  data.nextEntryId++;
+  data.entries.unshift(newEntry);
+
+  $image.src = 'images/placeholder-image-square.jpg';
+  $entryForm.reset();
+}
+
+var $photoInput = document.getElementById('photo-input');
+var $image = document.querySelector('img');
+var $entryForm = document.querySelector('#create-form');
+
+$photoInput.addEventListener('input', handleInput);
+$entryForm.addEventListener('submit', handleSubmit);
```

Comment:
I would place global variables and event listeners at the top of the document so they are immediately in view when somebody goes to view your code.